From 1a6bb7bcbdf58aa940132e630870af061dc88537 Mon Sep 17 00:00:00 2001
Message-ID: <1a6bb7bcbdf58aa940132e630870af061dc88537.1791178680.git.yann.morin@orange.com>
From: "Gary E. Miller" <gem@rellim.com>
Date: Tue, 7 Jul 2026 13:41:54 -0700
Subject: [PATCH] clients/gpsprof.py.in: Quote back ticks in title.

Someone could use the back tick to break out of the string and add
gnuplot commnds.

For issue 404.
Reported by: CuB3y0nd, and Wade Sparks <wsparks@vulncheck.com>
Upstream: https://gitlab.com/gpsd/gpsd/-/commit/1a6bb7bcbdf58aa940132e630870af061dc88537
CVE: CVE-2026-58459
Signed-off-by: Yann E. MORIN <yann.morin@orange.com>

---
 clients/gpsprof.py.in | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/clients/gpsprof.py.in b/clients/gpsprof.py.in
index 261e72665..202214769 100644
--- a/clients/gpsprof.py.in
+++ b/clients/gpsprof.py.in
@@ -198,8 +198,9 @@ class plotter(object):
         if 'subtype' in self.device:
             desc += "\\n%s" % self.device['subtype']
 
-        # escape ", and \n, for gnuplot, to not break strings
+        # escape ", `, and \n, for gnuplot, to not break strings
         desc = desc.replace('"', '\\042')
+        desc = desc.replace('`', '\\140')
         desc = desc.replace('\n', '')
 
         return desc
@@ -1268,8 +1269,9 @@ if __name__ == '__main__':
             options.title = plot.whatami()
         if options.subtitle:
             options.title += '\\n' + options.subtitle
-        # escape " for gnuplot, to not break strings
+        # escape ",  and`, for gnuplot, to not break strings
         options.title = options.title.replace('"', '\\042')
+        options.title = options.title.replace('"', '\\140')
         term_opts = ""
         truecolor_terms = ['png', 'sixelgd', 'wxt']
         if options.terminal in truecolor_terms:
-- 
2.43.0

